codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -447,9 +448,14 @@ class JsonSerializerGenerator(
 
     private fun RustWriter.jsonObjectWriter(context: MemberContext, inner: RustWriter.(String) -> Unit) {
         safeName("object").also { objectName ->
-            rust("let mut $objectName = ${context.writerExpression}.start_object();")
-            inner(objectName)
-            rust("$objectName.finish();")
+            if (context.shape.isTargetUnit()) {
+                rust("let $objectName = ${context.writerExpression}.start_object();")
+                rust("$objectName.finish();")
+            } else {
+                rust("let mut $objectName = ${context.writerExpression}.start_object();")
+                inner(objectName)
+                rust("$objectName.finish();")
+            }
```

Comment:
This can be simplified. If clippy complains, you can throw in an "allow unused mut" attribute.
```suggestion
            rust("let mut $objectName = ${context.writerExpression}.start_object();")
            if (context.shape.isTargetUnit()) {
                inner(objectName)
            }
            rust("$objectName.finish();")
```

Can you also add a comment explaining why the `isTargetUnit` check is necessary in the serializer?

Reply:
Thanks! Addressed in 4b9a355. I negated the condition at line 455 (in the suggested) for correctness.